### ISSUE TITLE
Merge documentation and community repos intoa repository called confidential-containers

### ISSUE THREAD

**Issue by wainersm** [confidential-containers/confidential-containers#83]
## Context

There is a task-force working on setting up [garm](https://github.com/cloudbase/garm) (#77) to manage the Github self-hosted runners on Azure. Those ephemeral Github runners on Azure should be used on CI/CD pipelines where the default github runners do not meet the requirements (e.g. when needed nested virtualization). At first, projects that should be leveraging those runners are the [operator](https://github.com/confidential-containers/operator) and [cloud-api-adaptor](https://github.com/confidential-containers/cloud-api-adaptor) but other CoCo projects may request their access in the future.

We need a place to host the existing code for that garm setup. @mkulke and I had already sent a pull request (https://github.com/confidential-containers/community/pull/82) with some draft code addressed to this community repository. Early today we chatted and concluded that the community repository may not be the best place to hold the CI/CD stuffs.

Therefore, I am opening this issue to request the creation of a new repository to host not just this garm setup but potentially new code related with CI/CD in general that crosses the boundaries of individual projects within the CoCo organization. More than a request, actually, I would like to use this issue to discuss options if any.     

## Project Description

The new repository should be used to host common CI/CD code/files.

Examples:
 * the github self-hosted runners setup code
 * nightly CD pipelines
 * release scripts

## Alignment with Confidential Containers

It should not necessarily become a new project. If we find an existing repository to put the code then even better. So it is up to discussion actually.

## GitHub repositories needed

https://github.com/confidential-containers/cicd is just a suggestion. Any better name?

Cc garm task-force (@surajssd, @mkulke, @jepio), @confidential-containers/tsc , @fidencio , @confidential-containers/peer-pod-maintainers  , @confidential-containers/operator-maintainers

**Comment by wainersm** [comment]
> Seems reasonable to me.
> 
> Will we need an instance of garm to be running somewhere? Do we have a place in mind?
>

Yes, at least one instance will be running on Azure itself. There is already some code to create the resources (resources groups, VM/container...etc), and that's what we are looking for a home.
 
> Do we plan to use this for Kata CI as well? Do we plan to use it for Azure resources only or also for some of the bare metal nodes (not that they really support autoscaling)?

@fidencio and I agreed that Kata CI would adopt whatever is implemented in CoCo, but we don't know whether to use the same garm instance or not. As the migration Jenkins -> Github is a big effort itself, the idea is to adopt garm later.

Perhaps it would be better to have a separate instance for Kata. Once we have the garm setup coded in code, there should be easy to deploy it on Kata CI.

The idea is to use garm only for Azure resources, i.e., instantiate VMs on-demand.

**Comment by dcmiddle** [comment]
> On community meeting today I will open to discuss an alternative:
> 
> * merge documentation and community into a single repository (confidential-containers) and have a CI folder

FYI for completeness .. past issue: 
https://github.com/confidential-containers/documentation/issues/27

**Comment by larrydewey** [comment]
I agree with @wainersm's proposal from the community meeting today. It makes a lot of sense to unify the repositories. Currently there is benefit with speed of development, but it seems to isolate some projects into silos.  While there are additionally good arguments surrounding the possible bottlenecks introduced in a monolithic repository, I feel that having everything centralized will help to drive a unified vision, and help reduce the points of failure during a release cycle. It will also help in the issue/pr triage process as everything will be in one location. That being said, I do think that implementing an incremental approach also would be important.

**Comment by wainersm** [comment]
Edited the title of this issue to reflect the actual agreement.

Apart from creating the new confidential-containers repository, it will be needed:
 * Migrate the community and documentation repositories as well as their issues
 * Archive the community and documentation repositories. Hopefully Github has a mean to redirect urls
 * Create a `cicd` (names suggestions are welcomed) folder to hold the CI/CD common code
 
What else?

@fitzthum @larrydewey @dcmiddle

**Comment by fitzthum** [comment]
What about oustanding PRs and issues in the community and documentation repos? Also worth noting that the CNCF maintainers table currently points to the community repo. We should probably update that even if we get redirects working.

**Comment by dcmiddle** [comment]
What would be the downside to putting the garm material in the Operator repo?

**Comment by magowan** [comment]
I think merging the community and documentation repos makes sense.
Equally happy for this to be a new project.

I think combining the community/documentation with the garm setup and general CI/CD code makes a good immediate goal.
In terms of other repositories, certainly peer pods operator and coco operator make more sense together now peer pods is part of the release. I think we should figure it out incrementally on a case by case basis.

Happy to volunteer to run with moving documentation contents/PRs/Issues etc across to a new "combined" repo.

**Comment by bpradipt** [comment]
> Edited the title of this issue to reflect the actual agreement.
> 
> Apart from creating the new confidential-containers repository, it will be needed:
> 
> * Migrate the community and documentation repositories as well as their issues
> * Archive the community and documentation repositories. Hopefully Github has a mean to redirect urls
> * Create a `cicd` (names suggestions are welcomed) folder to hold the CI/CD common code
> 

How about `infra` repo which will contain cicd, infrastructure as code (IaC) files, bots etc ?


> What else?
> 
> @fitzthum @larrydewey @dcmiddle

**Comment by fitzthum** [comment]
Ok, does someone want to try combining the repos or should we just create a CI or infra repo so that we are not blocked here?

**Comment by sameo** [comment]
Can we keep the community/doc content separated from the infra bits for now? An `infra` repo makes sense to me, as @bpradipt proposed.

**Comment by wainersm** [comment]
I'm okay with an infra repository for now that could be swallowed by the confidential-containers repository eventually. I will open another issue to track separated, keeping this open as we still want to merge docs/community.

**Comment by magowan** [comment]
Confirming Happy to own this and I will setup the confidential-containers repo.

**Comment by magowan** [comment]
Not found a way to move PR's and not sure it would work for the second repo I merge in even if we did .
But @fidencio pointed me towards https://docs.github.com/en/issues/tracking-your-work-with-issues/transferring-an-issue-to-another-repository?tool=webui to help move issues.

Looking to move documentation first before code freeze next week,
then leave merging community in until after release.

Open thought - Do we move closed issues or just open issues....?

**Comment by magowan** [comment]
Ok changed the title to better reflect where we have arrived - Merge documentation and community repos.
Rather than new repo as suggested by @dcmiddle I will merge one into the other.
In this case looking at open issues, PR's  etc I will merge Documentation into the community repo then rename the community repo to confidential-containers to better reflect its wider purpose.

Intention is to preserve commit history.

Check list (modified as I think of more)

- [ ] Combine maintainers lists for both repos
- [ ] create a PR to merge (preserving commit history) documentation into community
- [ ] Take wiki content from documentation and add to community wiki
- [ ] Rename the repo to confidential containers
- [ ] Copy across Issues from documentation to community
- [ ] PR to Fix up any links/references within confidential-containers project
- [ ] Create PRs as needed in other projects to fix any references.
- [ ] Request any open PRs against documentation gets recreated against confidnetial-containers repo


- [ ] Fix up any references with the

**Comment by magowan** [comment]
Useful link https://stackoverflow.com/questions/1425892/how-do-you-merge-two-git-repositories
Thanks @dcmiddle

**Comment by ariel-adam** [comment]
@magowan is this targeted for 0.7.0 or should we move it to 0.8.0 (I see a bunch of open issues on the checklist :-))?

**Comment by dcmiddle** [comment]
Think this was all completed long ago.
